Improve builder profit estimation by taking bid adjustments into account

ababino opened the pull request:
## 📝 Summary
1. Added a new script `cmd/core/bid-adjustments-backfill.go` to fetch bid adjustments from ultrasound relay. 
2. Amended builder profit query in `GetBuilderProfits` in `database/database.go`. Now, if the slot is a bid-adjusted-slot, the query will subtract the validator profit from coinbase_diff.

## ⛱ Motivation and Context
Builder profit is not accurate when bids are adjusted by ultrasound relay.

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test`
* [x] `go mod tidy`

metachris commented on database/database.go:
```diff
@@ -163,18 +163,37 @@ func (s *DatabaseService) GetBuilderProfits(since, until time.Time) (res []*Buil
+      round(PERCENTILE_DISC(0.5) WITHIN GROUP(ORDER BY CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as median_profit_per_block,
+      round(sum(CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as total_profit,
+      round(abs(sum(CASE WHEN coinbase_diff_eth < 0 THEN coinbase_diff_eth ELSE 0 END)), 4) as total_subsidies
+    FROM adjusted_payloads
```
This query worries me a little bit -- how much more expensive is it to run? Not sure if this could be a bottleneck. Do you have any data or estimates on query performance?

ababino commented:
@metachris I made some changes:
1. I optimize the new query by adding a better index
2. I added a backfill_adjustments script
3. I added the new commands to the README file

Regarding the performance of the new query, I run it with 1 day of data and these are the stats (I used `EXPLAIN ANALYZE`):
old query:
Execution Time: 27.580 ms
Peak Memory Usage: 913kB

new query:
 Execution Time: 59.042 ms
Peak Memory Usage: 1681kB

So, we are talking about 2x slower and 2x more memory. I don't know what are our hardware requirements. If we need more performance, we'd have to change the way we store the data so we move the computation from query time to data fetching time.

Let me know how we should proceed.
